src/ChangeStream.php

```diff
@@ -58,36 +48,53 @@ public function current()
 
     public function getId()
     {
-        return $this->csIt->getId();
+        return $this->csIt->getInnerIterator()->getId();
+    }
+
+    public function key()
+    {
+
     }
 
     public function next()
     {
         try {
             $this->csIt->next();
-            $this->resumeToken = $this->csIt->extract_resume_token($this->csIt->current());
-        } catch (ResumableException $e) {
+            $this->resumeToken = $this->extractResumeToken($this->csIt->current());
```

Comment:
I believe the original ChangeStreamIterator implementation captured a resume token from `current()`.

If this is extracting _after_ the calling `$this->csIt->next()`, I think it's possible for `$this->resumeToken` to be uninitialized if we hit an exception on the very first call to `next()`.

I actually do like the idea of capturing the resume token during iterator movement, in case the user decides to not call `current()` for some reason (maybe they're ignoring/skipping some results). In that case, it may make sense to extract at the following points:

 * When we first construct this class (in case the user never calls `rewind()`)
 * Each time `next()` is called

IIRC, the driver's cursor doesn't actually need an initial rewind when iterating with a wrapped IteratorIterator. You can test that to make sure, though. It still wouldn't hurt to call `IteratorIterator::rewind()` in our constructor, though.

On a separate note, if user's do call `rewind()`, they should only do so at the beginning of iteration. We can debate whether it would be better for `ChangeStream::rewind()` to conditionally throw if called after one or more calls to `ChangeStream::next()`, or to simply have `ChangeStream::rewind()` be a no-op.